PowerUp/app/src/main/java/powerup/systers/com/db/DatabaseHandler.java

```diff
@@ -134,10 +137,12 @@ public boolean setSessionId(CharSequence ScenarioName) {
         if (cursor.moveToFirst()) {
             // If the scene is already completed
             if (cursor.getInt(6) == 1) {
+                cursor.close();
                 return false;
             }
             SessionHistory.currSessionID = cursor.getInt(0);
```

Comment:
To implement my suggestion you would need to move this snippet to `else` part @codingblazer I would love to hear your thoughts as well.